### PR TITLE
feat: limit fetch to one concurrent request

### PR DESCRIPTION
**Pull request by leanmendoza**
# What? 
With the same mechanism of permissions, patch the fetch implementing a FIFO of fetchs.

# Why? 
https://github.com/decentraland/sdk/issues/50

### CODE REVIEW

File: packages/scene-system/sdk/Fetch.ts [hunk 1]
```diff
@@ -1,12 +1,59 @@
...
 
+export function createConcurrentFetch(originalFetch: FetchFunction, waitForNextUpdate: WaitForNextUpdateFunction) {
+  const fifoFetch: string[] = []
+  return async (resource: RequestInfo, init?: RequestInit | undefined): Promise<Response> => {
+    const hash = Math.random().toString()
+    fifoFetch.push(hash)
+
+    const resolving = () => fifoFetch.length > 0 && fifoFetch[0] !== hash
+
+    await waitForNextUpdate()
```
Comment (leanmendoza) [inline packages/scene-system/sdk/Fetch.ts:18]: avoiding to use the setTimeout here.

File: packages/scene-system/sdk/Fetch.ts [hunk 2]
```diff
@@ -1,12 +1,59 @@
...
   canUseFetch: boolean
   originalFetch: FetchFunction
   previewMode: boolean
   log(...a: any[]): void
 }
 
+export function createConcurrentFetch(originalFetch: FetchFunction, waitForNextUpdate: WaitForNextUpdateFunction) {
+  const fifoFetch: string[] = []
+  return async (resource: RequestInfo, init?: RequestInit | undefined): Promise<Response> => {
+    const hash = Math.random().toString()
```
Comment (leanmendoza) [inline packages/scene-system/sdk/Fetch.ts:13]: can be whatever unique id

File: packages/scene-system/sdk/Fetch.ts [hunk 3]
```diff
@@ -1,12 +1,59 @@
...
+      const result = await originalFetch(resource, init)
+
+      void waitForNextUpdate().then(() => {
+        if (resolving()) {
+          fifoFetch.shift()
+        }
+      })
+
+      return {
+        ...result,
```
Comment (leanmendoza) [inline packages/scene-system/sdk/Fetch.ts:38]: can implement other body getter

File: packages/scene-system/sdk/SceneRuntime.ts [hunk 4]
```diff
@@ -551,13 +551,31 @@ export abstract class SceneRuntime extends Script {
...
         })
-        const restrictedFetch = createFetch({
+        const safeFetch = createFetch({
           canUseFetch,
           originalFetch: this.originalFetch,
           previewMode: this.isPreview || unsafeAllowed,
           log: dcl.log
         })
 
+        type Resolver = (value: unknown) => void
```
Comment (leanmendoza) [inline packages/scene-system/sdk/SceneRuntime.ts:561]: this method (systemDidEnable) is getting longer and longer

**Review by gonpombo8 — APPROVED** [review]

**Pull request merged**